### PR TITLE
Add 1.2.0 release note.

### PR DESCRIPTION
**Pull request by jholdstock**

### CODE REVIEW

File: docs/release-note-1.2.0.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+- `Cache-Control` HTTP headers are used to prevent sensitive information being
+  cached by browsers
+- Error page no longer enables embedding malicious links via URL parameters
+
+### Terminology
+
+Various Decred terminology has changed since the last release:
+
+- "Stakepools" are now known as "Voting Service Providers"
+- The name Decred should no longer styled as "deCRED"
```
[the remark this answers is not in the document]
Comment (jholdstock) [reply docs/release-note-1.2.0.md]: Good catch. Fixed.

File: docs/release-note-1.2.0.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+### MySQL storage for HTTP sessions
+
+HTTP sessions in dcrstakepool are implemented using [gorilla/sessions](https://github.com/gorilla/sessions),
+which by default uses a file based storage solution for session cookies.
+This release replaces that implementation with a custom MySQL session store.
+dcrstakepool will create a new table `Session` in the MySQL database if it is not already present.
+This was done to address several security issues introduced by the default file storage.
+
+- A user who changes their password or email will now be logged out of all sessions on all devices.
+- Cookies stored in a browser can no longer be used obtain information after a user has logged out.
```
Comment (JoeGruffins) [inline docs/release-note-1.2.0.md]: Totally my fault but
"used to obtain"

File: docs/release-note-1.2.0.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+|`recaptchasecret` and `recaptchasitekey`|[#281](https://github.com/decred/dcrstakepool/pull/281)|CAPTCHA is now self-hosted and implemented using [dchest/captcha](https://github.com/dchest/captcha) instead of Google reCAPTCHA.|
+
+### Deprecated
+
+It is **recommended** to remove these config items from `dcrstakepool.conf`. dcrstakepool will ignore these config items and log a warning if they are set. They will be removed completely in the next release.
+
+|Config|PR|Reason|
+|------|--|------|
+|`wallethosts`, `walletusers`, `walletpasswords` and `walletcerts`|[#470](https://github.com/decred/dcrstakepool/pull/470)|dcrstakepool no longer contacts dcrwallet directly. All comms are now routed through stakepoold.|
+|`enablestakepoold`|[#398](https://github.com/decred/dcrstakepool/pull/398)|stakepoold is always required. dcrstakepool cannot functin without it.|
```
Comment (JoeGruffins) [inline docs/release-note-1.2.0.md]: "function"

**Comment by jholdstock** [comment]
Thanks Seth, Joe. I have addressed all of your comments.

I noticed Joe opened #510 which I think should definitely be included in the release, so let's finish that off and update this release note before merging it.

**Comment by s-ben** [comment]
@jholdstock still seeing the Autoreconnect PR open and being worked on. I take it v1.2.0 is not released yet?

**Comment by jholdstock** [comment]
Correct, 510 is the last piece to go in before release.

**Comment by JoeGruffins** [comment]
I did an upgrade from the older version to the newer one following the guide.  I tried to go as far back as possible. The earliest Stakepoold I could make run without changes was around dcrd and dcrwallet release-v1.3, but I was unable to buy tickets.  Since all dcrstakepools are currently on release-v1.4 or are forked off the network, I think it is safe to start from there.  I started with a blank db and a new wallet. Dcrstakepool was on commit https://github.com/decred/dcrstakepool/commit/296ef81bf3a715b69fb04ae35a75b566f0a2a52a 
The following is evidence:
Starting home page.
pic1

Made user and bought ticket.
pic2

And the guide...

4. Perform an upgrade of each back-end server, one at a time
   i. Stop stakepoold
pic3

   ii. Stop dcrwallet
pic17

   iii. Stop dcrd
pic18

   iv. Install latest dcrd binary and start
   (I just built)
pic4

pic9

   v. Install latest dcrwallet binary and start
pic5

pic10

   vi. Install latest stakepoold binary and start
pic6

pic11

   vii. Check log files for warnings or errors
none

5. Make required changes to dcrstakepool.conf (detailed [above](#config-changes))
pic12

6. Install latest dcrstakepool and start
pic13

pic14

New homepage
pic15

Ticket bought before updating
pic16

Conclustion: WORKS!

**Comment by jholdstock** [comment]
This is excellent, thanks for going through that Joe. 

#510 and newly added #519 still need to go in before release.

According to a [report from @girino:decred.org](https://matrix.to/#/!XijfkJigmGCmMkgTTl:decred.org/$156797243524684yfzhv:decred.org?via=decred.org&via=matrix.org), the low fee tickets feature is not working correctly

**Comment by jholdstock** [comment]
I have updated the release note with all recently merged issues. Just waiting for #527 to be merged and I think the release is ready to go :)

**Pull request merged**